diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0bd59d9..f67a729 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-cmake_minimum_required(VERSION 3.14)
+cmake_minimum_required(VERSION 3.18)
 get_directory_property(is-subproject PARENT_DIRECTORY)
 
 project(
@@ -37,8 +37,8 @@ if(WITH_ILP64)
   message(STATUS "Using LAPACK/BLAS ILP64 interface")
 endif()
 
-if(NOT TARGET "LAPACK::LAPACK")
-  find_package("custom-lapack" REQUIRED)
+if(NOT TARGET "multicharge::LAPACK")
+  find_package("multicharge-lapack" REQUIRED)
 endif()
 
 # Collect subprojects
@@ -77,8 +77,8 @@ target_link_libraries(
   "${PROJECT_NAME}-lib"
   PUBLIC
   "${lib-deps}"
-  "LAPACK::LAPACK"
-  "$<$<VERSION_LESS:${CMAKE_VERSION},3.20>:BLAS::BLAS>"
+  "multicharge::LAPACK"
+  "$<$<VERSION_LESS:${CMAKE_VERSION},3.20>:multicharge::BLAS>"
 )
 if(WITH_OpenMP)
   target_link_libraries(
diff --git a/config/CMakeLists.txt b/config/CMakeLists.txt
index 873a5ef..310812c 100644
--- a/config/CMakeLists.txt
+++ b/config/CMakeLists.txt
@@ -18,6 +18,7 @@ option(WITH_OpenMP "Enable support for shared memory parallelisation with OpenMP
 option(WITH_ILP64 "Enable support for ILP64 BLAS/LAPACK calls" FALSE)
 option(WITH_TESTS "Enable compilation of unit tests" TRUE)
 
+set(MULTICHARGE_BLAS ${BLA_VENDOR} CACHE STRING "BLAS vendor")
 
 if(NOT DEFINED "${PROJECT_NAME}-dependency-method")
   set(
diff --git a/config/cmake/Findcustom-blas.cmake b/config/cmake/Findcustom-blas.cmake
deleted file mode 100644
index d3760b0..0000000
--- a/config/cmake/Findcustom-blas.cmake
+++ /dev/null
@@ -1,30 +0,0 @@
-# This file is part of multicharge.
-# SPDX-Identifier: Apache-2.0
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-if ((BLA_VENDOR MATCHES ^Intel) OR (DEFINED ENV{MKLROOT}))
-  enable_language("C")
-endif()
-
-if(WITH_ILP64)
-  set(BLA_SIZEOF_INTEGER 8)
-endif()
-
-if(NOT BLAS_FOUND)
-  find_package("BLAS" REQUIRED)
-  if(NOT TARGET "BLAS::BLAS")
-    add_library("BLAS::BLAS" INTERFACE IMPORTED)
-    target_link_libraries("BLAS::BLAS" INTERFACE "${BLAS_LIBRARIES}")
-  endif()
-endif()
diff --git a/config/cmake/Findcustom-lapack.cmake b/config/cmake/Findcustom-lapack.cmake
deleted file mode 100644
index 0167a62..0000000
--- a/config/cmake/Findcustom-lapack.cmake
+++ /dev/null
@@ -1,35 +0,0 @@
-# This file is part of multicharge.
-# SPDX-Identifier: Apache-2.0
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-if ((BLA_VENDOR MATCHES ^Intel) OR (DEFINED ENV{MKLROOT}))
-  enable_language("C")
-endif()
-
-if(WITH_ILP64)
-  set(BLA_SIZEOF_INTEGER 8)
-endif()
-
-if(NOT LAPACK_FOUND)
-  find_package("LAPACK" REQUIRED)
-
-  if(NOT TARGET "BLAS::BLAS")
-    find_package("custom-blas" REQUIRED)
-  endif()
-
-  if(NOT TARGET "LAPACK::LAPACK")
-    add_library("LAPACK::LAPACK" INTERFACE IMPORTED)
-    target_link_libraries("LAPACK::LAPACK" INTERFACE "${LAPACK_LIBRARIES}" "BLAS::BLAS")
-  endif()
-endif()
diff --git a/config/cmake/Findmulticharge-blas.cmake b/config/cmake/Findmulticharge-blas.cmake
new file mode 100644
index 0000000..20a3a54
--- /dev/null
+++ b/config/cmake/Findmulticharge-blas.cmake
@@ -0,0 +1,53 @@
+# This file is part of multicharge.
+# SPDX-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if ((MULTICHARGE_BLAS MATCHES ^Intel) OR (DEFINED ENV{MKLROOT}))
+  enable_language("C")
+endif()
+
+if(WITH_ILP64)
+  set(BLA_SIZEOF_INTEGER 8)
+  set(_nvpl_int "_ilp64")
+else()
+  set(_nvpl_int "_lp64")
+  set(BLA_SIZEOF_INTEGER 4)
+endif()
+
+if(NOT MULTICHARGE_BLAS_FOUND)
+  if(MULTICHARGE_BLAS STREQUAL "NVPL")
+    find_package("nvpl_blas" REQUIRED)
+    set(MULTICHARGE_BLAS_FOUND TRUE)
+
+    if((BLA_THREAD STREQUAL "OMP") OR (BLA_THREAD STREQUAL "ANY"))
+      set(_nvpl_thread "_omp")
+    else()
+      set(_nvpl_thread "_seq")
+    endif()
+
+    add_library("multicharge::BLAS" INTERFACE IMPORTED GLOBAL)
+    target_link_libraries("multicharge::BLAS" INTERFACE "nvpl::blas${_nvpl_int}${_nvpl_thread}")
+  else()
+    find_package("BLAS" REQUIRED)
+    set(MULTICHARGE_BLAS_FOUND ${BLAS_FOUND})
+  
+    if(NOT TARGET "multicharge::BLAS")
+      add_library("multicharge::BLAS" INTERFACE IMPORTED GLOBAL)
+      target_link_libraries("multicharge::BLAS" INTERFACE "BLAS::BLAS")
+    endif()
+  endif()
+endif()
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(multicharge-blas DEFAULT_MSG MULTICHARGE_BLAS_FOUND)
diff --git a/config/cmake/Findmulticharge-lapack.cmake b/config/cmake/Findmulticharge-lapack.cmake
new file mode 100644
index 0000000..a1b9efb
--- /dev/null
+++ b/config/cmake/Findmulticharge-lapack.cmake
@@ -0,0 +1,59 @@
+# This file is part of multicharge.
+# SPDX-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if ((MULTICHARGE_BLAS MATCHES ^Intel) OR (DEFINED ENV{MKLROOT}))
+  enable_language("C")
+endif()
+
+if(WITH_ILP64)
+  set(BLA_SIZEOF_INTEGER 8)
+  set(_nvpl_int "_ilp64")
+else()
+  set(_nvpl_int "_lp64")
+  set(BLA_SIZEOF_INTEGER 4)
+endif()
+
+if(NOT TARGET "multicharge::BLAS")
+  find_package("multicharge-blas" REQUIRED)
+endif()
+
+if(NOT MULTICHARGE_LAPACK_FOUND)
+  if(MULTICHARGE_BLAS STREQUAL "NVPL")
+    find_package("nvpl_lapack" REQUIRED)
+    set(MULTICHARGE_LAPACK_FOUND TRUE)
+
+    if((BLA_THREAD STREQUAL "OMP") OR (BLA_THREAD STREQUAL "ANY"))
+      set(_nvpl_thread "_omp")
+    else()
+      set(_nvpl_thread "_seq")
+    endif()
+
+    add_library("multicharge::LAPACK" INTERFACE IMPORTED GLOBAL)
+    target_link_libraries("multicharge::LAPACK" INTERFACE "nvpl::lapack${_nvpl_int}${_nvpl_thread}" "multicharge::BLAS")
+  else()
+    find_package("LAPACK" REQUIRED)
+    set(MULTICHARGE_LAPACK_FOUND ${LAPACK_FOUND})
+  
+    if(NOT TARGET "multicharge::LAPACK")
+    add_library("multicharge::LAPACK" INTERFACE IMPORTED GLOBAL)
+    target_link_libraries("multicharge::LAPACK" INTERFACE "LAPACK::LAPACK" "multicharge::BLAS")
+    endif()
+  endif()
+
+
+endif()
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(multicharge-lapack DEFAULT_MSG MULTICHARGE_LAPACK_FOUND)
diff --git a/config/template.cmake b/config/template.cmake
index 3b4262a..57bcf32 100644
--- a/config/template.cmake
+++ b/config/template.cmake
@@ -12,8 +12,9 @@ if(NOT TARGET "@PROJECT_NAME@::@PROJECT_NAME@")
     find_dependency("OpenMP")
   endif()
 
-  if(NOT TARGET "LAPACK::LAPACK")
-    find_dependency("LAPACK")
+  set(MULTICHARGE_BLAS @MULTICHARGE_BLAS@)
+  if(NOT TARGET "multicharge::LAPACK")
+    find_dependency("multicharge-lapack")
   endif()
 
   if(NOT TARGET "mctc-lib::mctc-lib")
